src/transceivercontroller/DefaultTransceiverController.ts

```diff
@@ -402,4 +442,366 @@ export default class DefaultTransceiverController implements TransceiverControll
+          observerFunc(observer);
+        }
+      });
+    }
+  }
+
+  private setupAudioRedWorker(): void {
+    // @ts-ignore
+    const supportsRTCScriptTransform = !!window.RTCRtpScriptTransform;
+    // @ts-ignore
+    const supportsInsertableStreams = !!RTCRtpSender.prototype.createEncodedStreams;
+
+    if (supportsRTCScriptTransform) {
+      // This is the prefered approach according to
+      // https://github.com/w3c/webrtc-encoded-transform/blob/main/explainer.md
+      // but chrome doesn't seem to support it as of chrome 111.0
+      // Safari 16.3 seems to support this.
+      this.logger.info(
+        '[AudioRed] Supports encoded insertable streams using RTCRtpScriptTransform'
+      );
+    } else if (supportsInsertableStreams) {
+      this.logger.info('[AudioRed] Supports encoded insertable streams using TransformStream');
+    } else {
+      this.logger.warn('[AudioRed] Encoded insertable streams not supported');
+      return;
+    }
+
+    // Stringify the `RedundantAudioEncoder` class code and get the new name of the `RedundantAudioEncoder` class since
+    // its name is changed in the browser.
+    const redWorkerCode = RedundantAudioEncoder.toString();
+    const redClassName = redWorkerCode.match(/class\s*(\w+)\s*\{/)[1];
+
+    const blobParts: BlobPart[] = [redWorkerCode];
+    if (this.logger.getLogLevel() === LogLevel.DEBUG) {
+      blobParts.push(`${redClassName}.shouldLogDebug=1;`);
+    }
+    blobParts.push(`${redClassName}.shouldReportStats=1;`);
+    blobParts.push(`${redClassName}.initializeWorker();`);
+
+    this.audioRedWorkerURL = URL.createObjectURL(
+      new Blob(blobParts, {
+        type: 'application/javascript',
+      })
+    );
+    this.logger.info(`[AudioRed] Redundant audio worker URL ${this.audioRedWorkerURL}`);
+    this.audioRedWorker = new Worker(this.audioRedWorkerURL);
+    this.audioRedEnabled = true;
+
+    // We cannot use console.log in production code and we cannot
+    // transfer the logger object so we need the worker to post messages
+    // to the main thread for logging
+    this.audioRedWorker.onmessage = (event: MessageEvent) => {
+      /* istanbul ignore else */
+      if (event.data.type === 'REDWorkerDebugLog') {
+        this.logger.debug(event.data.log);
+      } /* istanbul ignore next */ else if (event.data.type === 'RedundantAudioEncoderStats') {
+        const redMetricReport = new RedundantAudioRecoveryMetricReport();
+        redMetricReport.currentTimestampMs = Date.now();
+        redMetricReport.ssrc = event.data.ssrc;
+        redMetricReport.totalAudioPacketsLost = event.data.totalAudioPacketsLost;
+        redMetricReport.totalAudioPacketsExpected = event.data.totalAudioPacketsExpected;
+        redMetricReport.totalAudioPacketsRecoveredRed = event.data.totalAudioPacketsRecoveredRed;
+        redMetricReport.totalAudioPacketsRecoveredFec = event.data.totalAudioPacketsRecoveredFec;
+        this.forEachRedMetricsObserver(observer => {
```

Comment:
`forEachRedMetricsObserver` is a private function and only seem to call here right? Can we simplify the code to:
```
  private forEachRedMetricsObserver(
  ): void {
    for (const observer of this.redMetricsObservers) {
      AsyncScheduler.nextTick(() => {
        /* istanbul ignore else */
        // Since this executes asynchronously, we need to check if the observer has not been removed.
        if (this.redMetricsObservers.has(observer) ) {
          observer.recoveryMetricsDidReceive(redMetricReport);
        }
      });
    }
  }
```

Reply:
Updated. I'm assuming you meant make it `forEachRedMetricsObserver(redMetricReport)`